### ISSUE TITLE
Replace amp-story-bookend with amp-story-social-share

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#6852]
## Task Description

As per https://github.com/ampproject/amphtml/issues/33218, we need to replace `<amp-story-bookend>` usage with the new `<amp-story-social-share>` component.

Mostly search+replace to change method name etc. in the `HTML` class, but the config is now also simpler.

**Comment by spacedmonkey** [comment]
I have created a draft PR - https://github.com/google/web-stories-wp/pull/6857

However in my testing, just switching out the tags didn't work. I think this is two fold. 

1. amp-story-social-share is not on the follow list of tags in [AMP_Allowed_Tags_Generated](https://github.com/ampproject/amp-wp/blob/bcd47bf5c4b45a8e4f4c2184441e4f4a6ccc40c8/includes/sanitizers/class-amp-allowed-tags-generated.php#L16). Without that, the tag is simple stripped. So we need to patch this upstream before we can continue. 
2. Even after I got the tag to render, it do not do anything. I believe this is because https://github.com/ampproject/amphtml/pull/33077 was only merge 3 days ago. Maybe this change as no gone out yet. I don't know.

If does a question in about analytic events as well that needs [answering](https://github.com/google/web-stories-wp/pull/6857/files#r597839626). 

Marking this ticket as blocked until external issues are fixed.

**Comment by swissspidy** [comment]
> I have created a draft PR - #6857
> 
> However in my testing, just switching out the tags didn't work. I think this is two fold.
> 
> 1. amp-story-social-share is not on the follow list of tags in [AMP_Allowed_Tags_Generated](https://github.com/ampproject/amp-wp/blob/bcd47bf5c4b45a8e4f4c2184441e4f4a6ccc40c8/includes/sanitizers/class-amp-allowed-tags-generated.php#L16). Without that, the tag is simple stripped. So we need to patch this upstream before we can continue.

Ah, yes, good point. Feel free to submit a PR to the amp-wp repo :-)

There is a Python script in the `bin` folder there. If setup is too complicated, perhaps @westonruter can do run this for us :-)

> 2. Even after I got the tag to render, it do not do anything. I believe this is because [ampproject/amphtml#33077](https://github.com/ampproject/amphtml/pull/33077) was only merge 3 days ago. Maybe this change as no gone out yet. I don't know.

[Opt-in to pre-release channels](https://github.com/ampproject/amphtml/blob/master/contributing/DEVELOPING.md#opting-in-to-pre-release-channels) to get the latest and greatest

> If does a question in about analytic events as well that needs [answering](https://github.com/google/web-stories-wp/pull/6857/files#r597839626).

Looks like those triggers are now superfluous and can be safely removed.

**Comment by westonruter** [comment]
I'm working on a spec update for the plugin. The validator spec structure has changed a bit so I need to make some changes to the generator.

**Comment by westonruter** [comment]
Please test this PR to ensure the spec changes are working for you: https://github.com/ampproject/amp-wp/pull/5998.

**Comment by spacedmonkey** [comment]
Thanks for the great work here @westonruter . I have updated my PR [here](https://github.com/google/web-stories-wp/pull/6857) to include that latest version of the plugin. Barring having to add `layout="nodisplay` to the tag, it now doesn't strip out the tag. 

It is also worth noting that I have had to change the format of the value in script tag meaning the filter may have to be changed [here](https://github.com/google/web-stories-wp/pull/6857/files#diff-be3612a52b2891d690f623d9052868cec381738b5a429283f025e401915a3c14L278-R279) as well. 

However, I am not seeing some PHPStan errors in the logs related to this update. I think these will need to fixed upstream, but are most likely not related to this change. See logs [here](https://github.com/google/web-stories-wp/runs/2165905114#step:10:11). 

Even through the tag is now being output in the markup, I could get anything to render. With a little hacking, I managed to change the script tag from `https://cdn.ampproject.org/v0.js` to `https://cdn.ampproject.org/v0.js?optin=beta`. However it is did not render. 

I think this PR is a little to early, we need to wait for this change to propagate. Any chance you could chase this for me with the amp team @swissspidy

**Comment by swissspidy** [comment]
> It is also worth noting that I have had to change the format of the value in script tag meaning the filter may have to be changed [here](https://github.com/google/web-stories-wp/pull/6857/files#diff-be3612a52b2891d690f623d9052868cec381738b5a429283f025e401915a3c14L278-R279) as well.

Are you sure about that?

According to https://github.com/ampproject/amphtml/issues/33218 the data structure didn't change at all.

How was it determined that it need to be changed? Were there any errors?

> However, I am not seeing some PHPStan errors in the logs related to this update. I think these will need to fixed upstream, but are most likely not related to this change. See logs [here](https://github.com/google/web-stories-wp/runs/2165905114#step:10:11).

The errors are all reported in `includes/AMP` and look valid to me. Probably undetected before, but now reported due to a dependency update. Example:

<img width="1259" alt="Screenshot 2021-03-22 at 15 17 45" src="https://user-images.githubusercontent.com/841956/112004076-c332cc80-8b21-11eb-8758-5e6195167160.png">

`DOMDocument::createElement` can indeed return `false`, yet we don't currently check for that.

So this needs fixing on our end. We can do that in the same PR.

> Even through the tag is now being output in the markup, I could get anything to render. With a little hacking, I managed to change the script tag from `https://cdn.ampproject.org/v0.js` to `https://cdn.ampproject.org/v0.js?optin=beta`. However it is did not render.

As mentioned above, [opt-in to pre-release channels](https://github.com/ampproject/amphtml/blob/master/contributing/DEVELOPING.md?rgh-link-date=2021-03-19T17%3A42%3A03Z#opting-in-to-pre-release-channels) to get the latest and greatest. This way you can load the beta channel.

> I think this PR is a little too early, we need to wait for this change to propagate. Any chance you could chase this for me with the amp team @swissspidy

A new release of AMP is pushed to all AMP pages every week on Tuesday. Once a change in the AMP repo is merged, it will typically take 1-2 weeks for the change to be live for all users.

That doesn't mean we can't proactively work on a PR like in this case :-) We can wait with a merge though.

**Comment by spacedmonkey** [comment]
I fixed the phpstan and phpmd issues, in the PR. These were related to update the amp plugin and tool kit. They are valid changed and are small. So I don't mind doing this. 

The issue why it wasn't working, was I tried beta and. not nightly. It worked in the nightly build. 

> According to ampproject/amphtml#33218 the data structure didn't change at all.

Change reverted. 

Here is it working 

![image](https://user-images.githubusercontent.com/237508/112054093-0d7d7300-8b4d-11eb-8547-6abd5771430d.png)
